environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaNetworkProvider.java

```diff
@@ -4,15 +4,24 @@
 
 import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
 import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.instance.InstanceGroupNetworkRequest;
 import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.network.NetworkRequest;
 
 public interface FreeIpaNetworkProvider {
 
-    NetworkRequest provider(EnvironmentDto environment);
+    NetworkRequest provider(EnvironmentDto environment, boolean multiAzRequired);
+
+    default InstanceGroupNetworkRequest providerByGroup(EnvironmentDto environment) {
```

Comment:
I know the first method was already existing code, but I find the naming somewhat confusing. Why not rename them as `network` and `networkByGroup`?